src/main/scala/com/cognite/spark/datasource/EventsRelation.scala

```diff
@@ -81,6 +84,23 @@ object UpdateEventItem {
     )
 }
 
+sealed trait RetrieveEvents
+case class EventId(id: Long) extends RetrieveEvents
+case class EventExternalId(externalId: String) extends RetrieveEvents
+
+object RetrieveEventCoders {
+  implicit val encodeRetrieveEvents: Encoder[RetrieveEvents] = new Encoder[RetrieveEvents] {
+    override def apply(a: RetrieveEvents): Json =
+      Json.obj(a match {
+        case EventId(id) => ("id", Json.fromLong(id))
+        case EventExternalId(id) => ("externalId", Json.fromString(id))
```

Comment:
This seems to be doing the same encoding as the default/inferred encoder would do...?

Reply:
The autoencoder would not properly encode the sealed trait class.
https://circe.github.io/circe/codecs/adt.html